### DISCUSSION TITLE
Improve Chart JS toolkit

### DISCUSSION THREAD

**Discussion by brandonkelly** [craftcms/cms#8406] [Ideas]
- [ ] Improve Docs for Axes
- [ ] Add a legends to the Area, Column, and Pie charts
- [ ] Make it possible to set width and height with JS settings

#### Legend:

- [ ] Make the width of legend items dynamic for horizontal legends
- [ ] Replace labels by arrow icons for pagination's prev/next buttons
- [ ] i18n & RTL support for pagination

#### Geo/Map:

- [ ] Try to get rid of the Datamap library in favor of D3's native Geo functionality
- [x] Translate country names for Map chart
- [x] Add more colors to the colors array
- [x] Add Pie, Area, and Column charts
- [x] Keep the ratio of the Map chart so that it doesn't get cropped

**Comment by putyourlightson** [comment]
Hey guys, just wondering if you expect to manage to add any further enhancements by the Craft 3 GA? 

Also, would it be possible to add a request for the ability to create stacked area charts? For example to display the following data set:

```
    var dataTable = new Craft.charts.DataTable({
        columns: [
            { label: "Date", type: 'date' },
            { label: "New Users", type: 'number' },
            { label: "Total Users", type: 'number' },
        ],
        rows: [
            ['2016-01-23', 5, 47],
            ['2016-01-24', 15, 62],
            ['2016-01-25', 10, 72],
            ['2016-01-26', 13, 85],
        ]
    });
```
